src/Distributed/distributed_grids.jl

```diff
@@ -0,0 +1,60 @@
+using Oceananigans.Grids
+using Oceananigans.Grids: cpu_face_constructor_x, cpu_face_constructor_y, cpu_face_constructor_z, pop_flat_elements
+
+
+@inline get_local_coords(c::Tuple         , nc, lc, index) = (c[1] + (index-1) * lc,    c[1] + index * lc)
+@inline get_local_coords(c::AbstractVector, nc, lc, index) = c[1 + (index-1) * nc : 1 + nc * index]
+
+@inline my_grid(grid::RectilinearGrid, x, y, z, size, halo) = RectilinearGrid(size = size, 
```

Comment:
The nomenclature "my" has always been confusing to me. Can we use a different scheme, like "local" and "global"?